### PR TITLE
Add Comments

### PR DESCRIPTION
**Pull request by JawadHossain**
Add comment feature.
Screen Shot 2021-10-19 at 5 56 22 PM
 
Todo: Like comment to be added in another pr.

### CODE REVIEW

File: code/api/views.py [hunk 1]
```diff
@@ -109,9 +110,51 @@ def get(self, request, author_id, post_id):
...
-        return HttpResponse("This is the authors/aid/posts/pid/comments/ endpoint")
+        # Send all comments 
+        # TODO: Update json response and data serialization to match spec
+        try:
+            comments = Comment.objects.filter(post=post_id).order_by('-pub_date')
+            
+            serializedComments = []
+            for comment in comments:
+                serialized_obj = serializers.serialize('json', [ comment, ])
+                serializedComments.append(serialized_obj)
```
Comment (debecerra) [inline code/api/views.py:126]: As mentioned in the meeting, you can take a look at the work I did for authors and followers for how to correctly format the response according to the spec

File: code/api/views.py [hunk 2]
```diff
@@ -109,9 +110,51 @@ def get(self, request, author_id, post_id):
...
+            return HttpResponseServerError()
+          
+        return JsonResponse(serializedComments, safe=False)
+
+    def post(self, request, author_id, post_id):
+        # check if authenticated
+        if (not request.user):
+            return HttpResponseForbidden()
+
+        comment = request.POST.get('comment')
```
Comment (debecerra) [inline code/api/views.py:138]: Can you take a look at the way I parsed JSON input in the inbox/ endpoint and compare? I'm not saying anything is wrong with this, I'm just not sure if the API is supposed to receive data using request.POST.get(param), or the data should be stored as JSON and read from request.body. Let me know if it still doesn't make sense.

File: code/socialDistribution/templates/tagtemplates/author.html [hunk 3]
```diff
@@ -1,45 +1,64 @@
 {% load static %}
-<div class="author-tile card shadow bg-white m-5 h-100">
+<div class="author-tile card shadow bg-white m-2 h-100 {% if not comment_card %} m-5 {% endif %}">
```
Comment (debecerra) [inline code/socialDistribution/templates/tagtemplates/author.html:2]: Does it really make cohesive sense to use the author card for comments as well? To me, it feels like comment tiles and author tiles have a different purpose and meaning. Just thinking about code maintainability. We can always refactor later too

Comment (JawadHossain) [reply code/socialDistribution/templates/tagtemplates/author.html:2]: Was easy to reuse the card but I will create a comment-card if it grows too big.

File: code/api/views.py (hunk 1 above)
Comment (JawadHossain) [reply code/api/views.py:154]: thanks! create #89  for this

File: code/api/views.py (hunk 2 above)
Comment (JawadHossain) [reply code/api/views.py:166]: I think we will read it from the request body. Thanks! Will make sure to fix here #89

Comment (olivier-vadiaval) [reply code/api/views.py:166]: Can we add a check for empty comments here? It seems that the user is able to post blank comments at the moment.

Comment (JawadHossain) [reply code/api/views.py:166]: Yes, will do!

Comment (JawadHossain) [reply code/api/views.py:166]: 383282d3d3455e7d5fbee38772595495180fe1a5
Done.

**Pull request merged**